### PR TITLE
Feedback

### PR DESCRIPTION
**Pull request by life-efficient**
:wave:! GitHub Classroom created this pull request as a place for your teacher to leave feedback on your work. It will update automatically. **Don’t close or merge this pull request**, unless you’re instructed to do so by your teacher.

In this pull request, your teacher can leave comments and feedback on your code. Click the **Subscribe** button to be notified if that happens.

Click the **Files changed** or **Commits** tab to see all of the changes pushed to `master` since the assignment started. Your teacher can see this too.

<details>
<summary>
<strong>Notes for teachers</strong>
</summary>

Use this PR to leave feedback. Here are some tips:

- Click the **Files changed** tab to see all of the changes pushed to `master` since the assignment started. To leave comments on specific lines of code, put your cursor over a line of code and click the blue **+** (plus sign). To learn more about comments, read “[Commenting on a pull request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request)”.
- Click the **Commits** tab to see the commits pushed to `master`. Click a commit to see specific changes.
- If you turned on autograding, then click the **Checks** tab to see the results.
- This page is an overview. It shows commits, line comments, and general comments. You can leave a general comment below.

For more information about this pull request, read “[Leaving assignment feedback in GitHub](https://classroom.github.com/help/leaving-feedback-in-github)”.
</details>

Subscribed: @42FraMeWork

### CODE REVIEW

File: hyper_search.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
+
+
+from sklearn.model_selection import train_test_split
+import numpy as np
+
+def max_depth_search(model, X, y, cycles=20, depth_list=None, verbose=False):
+    
+    if depth_list is None:
+        depth_list = [1, 3, 6, 10, None]
```
Comment (life-efficient) [inline hyper_search.py:9]: Set this as a default argument instead of setting the default to None and then also having an extra if statement

File: hyper_search.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+    highest_acc = 0
+    opt_depth = 1
+
+    for depth in depth_list:
+        
+        accuracy_list = []
+        for _ in range(cycles):
+
+            X_train, X_validation, y_train, y_validation = train_test_split(X, y, train_size=0.8)
```
Comment (life-efficient) [inline hyper_search.py:19]: all your models should have the same validation set. put this in the main file and pass the val set to each search

File: hyper_search.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    if depth_list is None:
+        depth_list = [1, 3, 6, 10, None]
+
+    highest_acc = 0
+    opt_depth = 1
+
+    for depth in depth_list:
+        
+        accuracy_list = []
+        for _ in range(cycles):
```
Comment (life-efficient) [inline hyper_search.py:17]: perhaps num_models_trained would be a more verbose name?

File: hyper_search.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+    return opt_lr, highest_acc
+
+def max_features_search(model, X, y, cycles=20, max_features_list=None, max_depth=None, verbose=False):
+    
+    if max_features_list is None:
+        max_features_list = [i + 1 for i in range(X.shape[1] -1)]
+        max_features_list.append('sqrt')
+        max_features_list.append('log2')
+        max_features_list.append('auto')
```
Comment (life-efficient) [inline hyper_search.py:93]: What's this?

File: hyper_search.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    for features in max_features_list:
+        
+        accuracy_list = []
+        for _ in range(20):
+            X_train, X_validation, y_train, y_validation = train_test_split(X, y, train_size=0.8)
+            rf_c = model(max_features= features, max_depth=max_depth)
+            rf_c.fit(X_train, y_train)
+            y_hat = rf_c.predict(X_validation).round()
+
+            acc = 1 - np.mean(y_hat != y_validation)
```
Comment (life-efficient) [inline hyper_search.py:107]: why not just take the mean of where they are equal?

File: project.ipynb [hunk 6]
```diff
@@ -0,0 +1,329 @@
+{
```
Comment (life-efficient) [inline project.ipynb:1]: <p>Would be good to get a description of the dataset here, and what you notice about it on first glance. </p><p>Some things I would include would be:</p><ul><li>num features</li><li>num examples</li><li>proportion of examples in test vs train set</li><li>is this a regression or classification problem? or something in between?</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AI-Core-Assignments/project-1-42FraMeWork/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (life-efficient) [inline project.ipynb:1]: <p>why are some of your charts missing axis labels?</p><p> </p><p>use more verbose variable name than. What does <code>reg</code> mean? Regular?</p><p> </p><p>Are your features on the same order of magnitude? Might you be able to get away without normalising for certain models? Why?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AI-Core-Assignments/project-1-42FraMeWork/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (life-efficient) [inline project.ipynb:1]: <p>How can wesee that from the data? And why is the spread relevant to that?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AI-Core-Assignments/project-1-42FraMeWork/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (life-efficient) [inline project.ipynb:1]: <p>No data preprocessing? Justification for not normalising?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AI-Core-Assignments/project-1-42FraMeWork/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (life-efficient) [inline project.ipynb:1]: <p>Quantify how effective. </p><p> </p><p>8% spread over how many training test.</p><p> </p><p>NO VALIDATION SET?!?!?! Why?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AI-Core-Assignments/project-1-42FraMeWork/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (life-efficient) [inline project.ipynb:1]: <p>Highly useful? How useful is that?</p><p> </p><p>1 level of quality away... interesting metric to track. Explain why this is very important. You claim classification works well from observing it's accuracy (perfectly predicting the correct label), but if an example has a label of 5 is it better to predict of 4 or 1?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AI-Core-Assignments/project-1-42FraMeWork/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>
